serverlessworkflow/modules/ROOT/pages/eventing/working-with-openapi-callbacks.adoc

```diff
@@ -0,0 +1,124 @@
+
+The `serverless-workflow-callback-events-over-http-quarkus` application is initiated with the following request to `http://localhost:8080/callback`:
+[code,json]
+----
+{
+  "message": "Hello"
+}
+----
+Once the workflow is started, it makes an external service call with the callback URL and the workflow instance ID in the request body to link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus/callback-event-service[callback-event-service]. Then, as configured in the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus/callback-workflow/src/main/resources/specs/callback.yaml[OpenAPI] file, the callback URL is invoked to send a CloudEvent to the workflow.
+
+Once the `wait` type CloudEvent is received by the callback-workflow-service, the workflow moves to the next state and ends successfully. The following figure shows the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus[`serverless-workflow-callback-events-over-http-quarkus`] image:
+
+image::eventing/openapi-callback.png[]
+
+To use the OpenAPI callback in a workflow, the OpenAPI YAML file is configured with the callback as specified in the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus/callback-workflow/src/main/resources/specs/callback.yaml[OpenAPI] file.
+
+To use the Callback state in a workflow, first CloudEvent type `wait` is declared that the workflow uses. Following is an example of CloudEvents declaration in a workflow definition:
+
+.Example of CloudEvent declaration in a workflow definition
+[code,json]
+----
+ "events": [
+    {
+      "name": "waitEvent",
+      "source": "",
+      "type": "wait"
+    }
+----
+
+After that, a Callback state is declared, and it waits for a CloudEvent with the `wait` type. Following is an example of declaring a Callback state that handles the `wait` type CloudEvent:
+
+.Example of a Callback State declaration handling the `wait` CloudEvent
+[code,json]
+----
+ {
+      "name": "waitForEvent",
+      "type": "callback",
+      "action":
+        {
+        "functionRef": {
+          "refName": "callBack",
+          "arguments": {
+            "uri": "http://localhost:8080/wait",
+            "processInstanceId": "$WORKFLOW.instanceId"
+          }
+        }
+      },
+      "eventRef": "waitEvent",
+      "transition": "finish"
+    }
+----
+*TIP:* Please refer xref:service-orchestration/configuring-openapi-services-endpoints.adoc[configure openapi service endpoints] document to set the URL dynamically using an environment variable.
```

Comment:
For TIP please use:
https://github.com/kiegroup/kogito-docs/blame/eaef87f88363074d330798e2bd6fe25ba897dcdb/serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc#L44-L47